packages/flutter/lib/src/widgets/platform_view.dart

```diff
@@ -580,3 +580,50 @@ class _UiKitPlatformView extends LeafRenderObjectWidget {
     renderObject.updateGestureRecognizers(gestureRecognizers);
   }
 }
+
+/// A helper widget provides the implementer of a new platform view certain utilities.
+///
+/// The widget is responsible to handle painting, gestures and semantics.
+class PlatformViewSurface extends LeafRenderObjectWidget {
+
+  /// Construct a `PlatformViewSurface`.
+  ///
+  /// The [id] and the [controller] must not be null.
+  /// By default, this widget will create a `PlatformViewLayer` with rect = size & offset and the same id of this `PlatformViewSurface`in the layer tree. If you want to use
+  /// a different type of layer (e.g.Android), or a different configuration(e.g. different sizes), implement [customLayerFactory] to have a customized layer created instead.
+  const PlatformViewSurface({
+    @required this.id,
+    @required this.controller,
+    PlatformViewRenderBoxLayerFactory customLayerFactory,
+  }) : assert(id != null),
+       assert(controller != null),
+      _customLayerFactory = customLayerFactory;
+
+  /// The id of the platform view that is associate with this `PlatformViewSurface`.
```

Comment:
What are the valid values for this?  There's a range, right?

What is this used for?  Add prose to explain that please.

Reply:
Also, the ref to a class should be `[PlatformViewSurface]`, but it probably isnt' needed here anyway.

Reply:
done